dist/pythonlibs/testrunner/spawn.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2018-19 Freie Universität Berlin
+#               2017 Cenk Gündoğan <cenk.guendogan@haw-hamburg.de>
+#               2016 Kaspar Schleiser <kaspar@schleiser.de>
+#               2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import pexpect
+import signal
+import sys
+import subprocess
+import time
+from traceback import extract_tb, print_tb
+
+PEXPECT_PATH = os.path.dirname(pexpect.__file__)
+RIOTBASE = (os.environ.get('RIOTBASE') or
+            os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                         "..", "..", "..")))
+
+# Setting an empty 'TESTRUNNER_START_DELAY' environment variable use the
+# default value (3)
+MAKE_TERM_STARTED_DELAY = int(os.environ.get('TESTRUNNER_START_DELAY') or 3)
+
+
+def list_until(l, cond):
+    return l[:([i for i, e in enumerate(l) if cond(e)][0])]
+
+
+def find_exc_origin(exc_info):
+    pos = list_until(extract_tb(exc_info),
+                     lambda frame: frame[0].startswith(PEXPECT_PATH)
+                     )[-1]
+    return (pos[3], os.path.relpath(os.path.abspath(pos[0]), RIOTBASE), pos[1])
+
+
+def setup_child(timeout=10, echo=True, traceback=False):
```

Comment:
`traceback` is unused as argument here.

To include also my needs for the functions.

I would like to change the interface to take `logfile=None` instead of `echo=True`. 
This will allow me to use something else than `sys.stdout`. Also I remember @jcarrano telling me that having boolean arguments are usually a sign of a bad pattern.

Also, it would help me if there could be a `spawnclass=pexpect.spawnu` argument. Because I currently need to overwrite it in my own wrappers. It shows a need for a class there, but at least would make it usable directly for me.

I also see again that the `env = os.environ.copy()` does nothing here too and could be removed if you want.